### PR TITLE
perf: implement pagination for GraphQL review threads (#63)

### LINKED ISSUE #63: perf: implement pagination for review threads in GraphQL queries
## Overview

From PR review comment by @copilot-pull-request-reviewer on PR #62

## Original Comment

> The query uses `first: 100` to limit review threads. If a PR has more than 100 review threads, some Copilot comments might be missed. Consider either documenting this limitation or implementing pagination to handle PRs with many review threads.

## Context

- **PR:** #62
- **Files:** `scripts/pr-review-poll.sh:339`, `.claude/commands/pr-review.md:348,595`

## Current Behavior

The GraphQL queries for fetching review threads use `first: 100` which limits results to 100 threads. For PRs with more than 100 review threads, some threads could be missed.

## Proposed Work

Implement pagination using `pageInfo { hasNextPage endCursor }` to handle PRs with many review threads:

1. Fetch threads in batches using cursor-based pagination
2. Accumulate results across all pages
3. Handle the case where cursor becomes null or API returns errors

## Acceptance Criteria

- [ ] Implement pagination loop for `check_copilot_pending_comments` function
- [ ] Update documentation examples with pagination pattern
- [ ] Add tests for large PR scenarios if possible

---

*Created from PR review: https://github.com/goffity/claude-km-skill/pull/62*

### PR DESCRIPTION
**Pull request by goffity**
## Summary

Implement cursor-based pagination for GraphQL queries to handle PRs with >100 review threads.

## Changes Made

- `scripts/pr-review-poll.sh` - Rewrite `check_copilot_pending_comments()` with pagination
- `.claude/commands/pr-review.md` - Update documentation with pagination examples

## Features

- Uses `pageInfo { hasNextPage endCursor }` for cursor-based pagination
- Loops through all pages accumulating counts
- Handles edge cases (null cursor, API errors)
- Updates `get_thread_id()` and `resolve_all_copilot_threads()` documentation

## Testing

| Test | Status |
|------|--------|
| Bash Syntax Validation | ✅ Passed (12/12) |
| Scripts Executable | ✅ Passed (12/12) |
| Code Review | ✅ Passed (0 critical) |

## Related Issues

Fixes #63

### CODE REVIEW

File: .claude/commands/pr-review.md [hunk 1]
```diff
@@ -335,32 +335,61 @@ tags: [relevant-tags]
...
                     }
                 }
             }
-        }
-    ' -f owner="$owner" -f repo="$repo" -F pr="$pr_number"
+        ' -f owner="$owner" -f repo="$repo" -F pr="$pr_number" $cursor_arg)
+
+        # Accumulate threads
+        local page_threads=$(echo "$result" | jq '.data.repository.pullRequest.reviewThreads.nodes')
+        all_threads=$(echo "$all_threads $page_threads" | jq -s 'add')
```
[the remark this answers is not in the document]
Comment (goffity) [reply .claude/commands/pr-review.md]: Fixed in 1937327\! Updated to use null-safe jq operations:
```bash
# Added // [] for null safety
local page_threads=$(echo "$result" | jq '.data.repository.pullRequest.reviewThreads.nodes // []')
# Use jq -n with --argjson for robust JSON merging
all_threads=$(jq -n --argjson all "$all_threads" --argjson page "$page_threads" '$all + $page')
```
